[WIP] tiled synthesis simulation

Starnite opened the pull request.

Starnite committed:
Downloaded PB2 sequences

Starnite committed:
Merge branch 'master' into vz-tiled-synthesis

Starnite committed:
Code for generating uniform tiles from historical PB2 variants

ericmjl commented:
Good start! Might need you to walk me through your code in person. Three things that I'm currently checking for - the tiles are overlapping, and that they're in DNA rather than amino acids, and that both the forward and reverse strands are available. 

I also found some papers that might give some inspiration:

- [Multiplex pairwise assembly](http://nar.oxfordjournals.org/content/early/2015/11/06/nar.gkv1177.long)
- [Dial-out PCR](https://www.ncbi.nlm.nih.gov/pubmed/22886093)

Give them a read-through, and let's talk about them in the lab (whenever you're in).

Starnite commented:
Do you remember what search criteria we used when going through the database? I’m searching for 2011-2016, human H3N2 type A PB2, but I’m getting a lot more results than I downloaded before break.

Starnite committed:
Fixed not overlapping error

ericmjl commented:
I don't remember the search parameters, but you could try removing duplicate sequences (it's one of the additional options available). Good reminder this time round to record the search params down too. :smile:

Starnite committed:
DNA sequences correlating with previously downloaded protein sequences

Search criteria: Human H3N2 Type A, 2011-2016, complete segments only,
all geographical origins, remove duplicates.

Starnite committed:
Updated code with f/r DNA sequences

ericmjl commented:
Great work!

Here's another consideration - I think there's many more unique nucleotide sequences than unique protein sequences. Makes sense, because of the degenerate 3rd codon. If you were to use a reverse translation (AA -> DNA) of the protein sequences vs. straight-up using known nucleotide sequences, what's the difference in the number of unique tiles that have to be synthesized?

Starnite commented:
I originally looked for unique nucleotide sequences in the database (there are nearly 3000), but my thinking was that for the most part, the polymorphisms wouldn’t really matter unless it resulted in a translation to a different amino acid. If I did reverse translation, I could either do every possible codon combination, or the most common codon for influenza.

ericmjl commented:
True, nucleotide degeneracy in the 3rd codon position might not matter for protein expression, but from an on-chip DNA synthesis standpoint, it very likely will matter. Each degenerate sequence is a new strand that needs to be made, and there's an upper-limit (not sure how much right now, but assumed to be on the order of 10K-100K) of the number of unique nucleotide strands that can be synthesized on one chip.

Starnite commented:
I might be misunderstanding something here – wouldn’t there be far too many possibilities if we took into account every degeneracy for every amino acid in the protein?

I will be back tomorrow afternoon – snow got me stuck in NY for an extra day. See you soon!

ericmjl commented:
You're right!! I was hoping you'd highlight this independent of myself
telling you that point :D.

What would be the engineering implications of this point? If the goal is
"make protein sequence (in fragments) while minimizing number of nucleotide
fragments required", how would you go about designing the oligos to be
synthesized on-chip?

On Sat, Jan 7, 2017 at 11:07 AM, Starnite <notifications@github.com> wrote:

> I might be misunderstanding something here – wouldn’t there be far too
> many possibilities if we took into account every degeneracy for every amino
> acid in the protein?
>
> I will be back tomorrow afternoon – snow got me stuck in NY for an extra
> day. See you soon!
>
> From: Eric Ma <notifications@github.com>
> Reply-To: ericmjl/protein-systematic-characterization <
> reply@reply.github.com>
> Date: Wednesday, January 4, 2017 at 2:01 PM
> To: ericmjl/protein-systematic-characterization <protein-systematic-
> characterization@noreply.github.com>
> Cc: Jiwei Zhong <vivzhong@mit.edu>, Author <author@noreply.github.com>
> Subject: Re: [ericmjl/protein-systematic-characterization] [WIP] tiled
> synthesis simulation (#61)
>
>
> True, nucleotide degeneracy in the 3rd codon position might not matter for
> protein expression, but from an on-chip DNA synthesis standpoint, it very
> likely will. Each degenerate sequence is a new strand that needs to be
> made, and there's an upper-limit (not sure how much right now, but assumed
> to be on the order of 10K-100K) of the number of unique nucleotide strands
> that can be synthesized on one chip.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub<https://github.com/
> ericmjl/protein-systematic-characterization/pull/61#issuecomment-270456187>,
> or mute the thread<https://github.com/notifications/unsubscribe-auth/
> ANrO2aEdcgMrtPeG06KxdQ2MnZmlPshvks5rO-yWgaJpZM4LBsDb>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ericmjl/protein-systematic-characterization/pull/61#issuecomment-271092210>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACgnjqqYSxwRnbrUlYjornuRGud05VBxks5rP7hVgaJpZM4LBsDb>
> .
>

Starnite committed:
Downloaded all unique DNA sequences

Starnite commented:
The two articles I mentioned today:

http://www.nature.com/nature/journal/v432/n7020/full/nature03151.html
http://www.nature.com/nbt/journal/v28/n12/full/nbt.1716.html

See you tomorrow!

Starnite committed:
Changed some file names

Starnite committed:
Fixed previous code errors

Namely, trimmed sequences to ORF if they weren’t already. Not
completely working yet.

Starnite committed:
Fixed ORF trimming

Starnite committed:
Added 5-base barcode and overlap for assembly

Starnite committed:
Optimize Tm of tile overlaps

Overlaps are adjusted such that the Tm is within 4C of the highest Tm

Starnite committed:
Merge branch 'master' into vz-tiled-synthesis

ericmjl commented:
Progress looking good. I think a good next step is to refactor the code into a Python module that you can import elsewhere, and write a few small-scale test cases that can be used to check for correctness in implementation.

Starnite committed:
wrote tests for generate_tiles.py

Starnite committed:
Added test file

Starnite committed:
Goals for Feb

ericmjl commented on experiments/20170207-tiled-synthesis-computation/Goals-February.md:
```diff
@@ -0,0 +1,9 @@
+## Goals for the month of February
```
@Starnite, separate PR for goals please! :smile:

ericmjl commented:
@Starnite: can you demo your code for me this Thursday, say in the morning? It's so that I can get a better feel for it.

Starnite committed:
Merge branch 'master' into vz-tiled-synthesis

Starnite committed:
Revert "Goals for Feb"

This reverts commit 51fa18887e0131acd926ba6dd38e22be802d41d7.

Starnite commented:
Sure! Would 10am work for you?

ericmjl commented:
Let's do that. What's your calendar email address? MIT or Gmail?

On Tue, Feb 7, 2017 at 8:55 AM, Starnite <notifications@github.com> wrote:

> Sure! Would 10am work for you?
>
> From: Eric Ma <notifications@github.com>
> Reply-To: ericmjl/protein-systematic-characterization <
> reply@reply.github.com>
> Date: Tuesday, February 7, 2017 at 7:50 AM
> To: ericmjl/protein-systematic-characterization <protein-systematic-
> characterization@noreply.github.com>
> Cc: Jiwei Zhong <vivzhong@mit.edu>, Mention <mention@noreply.github.com>
> Subject: Re: [ericmjl/protein-systematic-characterization] [WIP] tiled
> synthesis simulation (#61)
>
>
> @Starnite<https://github.com/Starnite>: can you demo your code for me
> this Thursday, say in the morning? It's so that I can get a better feel for
> it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub<https://github.com/
> ericmjl/protein-systematic-characterization/pull/61#issuecomment-277989803>,
> or mute the thread<https://github.com/notifications/unsubscribe-auth/
> ANrO2RrN0wwEYXAohBJ1cyZm5Kd1iMihks5raGiTgaJpZM4LBsDb>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ericmjl/protein-systematic-characterization/pull/61#issuecomment-278006166>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACgnjhYmzGgRgGQc8P37URkRtTnh-sQzks5raHfQgaJpZM4LBsDb>
> .
>

Starnite commented:
Gmail – vivianzhong06@gmail.com

Starnite committed:
Merge branch 'master' into vz-tiled-synthesis

Starnite committed:
Fixed EOL error

Starnite committed:
Fixed Tm optimization so that it preserves original sequence in overlaps

Starnite committed:
Fixed tm error

Starnite committed:
Playing around with networks

Starnite committed:
tiled synthesis simulation

Reverted to non-Tm-optimized overlaps for trying out graphs with tiles.
The nodes and edges all seem to have been added appropriately, except
all_simple_paths can’t seem to find any paths between the first and
last tiles.

Starnite committed:
Replaced nested for loops with itertools

Starnite committed:
Added some documentation and code for calculating the division of sequences into tiles

Starnite committed:
Finished code for set 30-base overlaps, perfect match

Starnite commented:
@ericmjl For 30-bp overlaps and threshold = 1, and testing with 5 sequences to start with, I have 11 re-constructed sequences that are present in the original pool, out of 1896 total.

ericmjl commented:
@Starnite Hmm... a bit confused here, how do we start with 5 sequences and end up with 11 reconstructed out of the original 5?

Starnite commented:
Many tiles are identical, so there are duplicates of certain sequences – so they’re different paths, but because the nodes are the same, the sequence is the same.

ericmjl commented:
Got it. This is interesting, it means that the pool of non-original
sequences from the tiles is quite large, under the conditions where one
mismatch per tile is allowed. This is good information; it'll elevate the
importance of PacBio sequencing + barcoding in rationally arraying the
protein variants.

Can you do me a favour and test the code on influenza neuraminidase? Use
global H3N2 neuraminidase sequences from the IRD, starting from year 2000
to present. Under the assumption of only perfect matches, how many
originals out of the space of total possibles are there?

Starnite commented:
Sure, will do!

From: Eric Ma <notifications@github.com>
Reply-To: ericmjl/protein-systematic-characterization <reply@reply.github.com>
Date: Tuesday, March 14, 2017 at 11:28 AM
To: ericmjl/protein-systematic-characterization <protein-systematic-characterization@noreply.github.com>
Cc: Vivian Zhong <vivzhong@mit.edu>, Mention <mention@noreply.github.com>
Subject: Re: [ericmjl/protein-systematic-characterization] [WIP] tiled synthesis simulation (#61)

Got it. This is interesting, it means that the pool of non-original
sequences from the tiles is quite large, under the conditions where one
mismatch per tile is allowed. This is good information; it'll elevate the
importance of PacBio sequencing + barcoding in rationally arraying the
protein variants.

Can you do me a favour and test the code on influenza neuraminidase? Use
global H3N2 neuraminidase sequences from the IRD, starting from year 2000
to present. Under the assumption of only perfect matches, how many
originals out of the space of total possibles are there?


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/ericmjl/protein-systematic-characterization/pull/61#issuecomment-286457535>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ANrO2b95MzTEERSrquS5bdwuDEALU-Y3ks5rlrIFgaJpZM4LBsDb>.

Starnite commented:
Hi Eric,

With tile lengths of 210 (30bp overlap on each end included), and testing with 10 sequences, I get 108 original out of 61666 total.



From: Eric Ma <notifications@github.com>
Reply-To: ericmjl/protein-systematic-characterization <reply@reply.github.com>
Date: Tuesday, March 14, 2017 at 11:28 AM
To: ericmjl/protein-systematic-characterization <protein-systematic-characterization@noreply.github.com>
Cc: Vivian Zhong <vivzhong@mit.edu>, Mention <mention@noreply.github.com>
Subject: Re: [ericmjl/protein-systematic-characterization] [WIP] tiled synthesis simulation (#61)

Got it. This is interesting, it means that the pool of non-original
sequences from the tiles is quite large, under the conditions where one
mismatch per tile is allowed. This is good information; it'll elevate the
importance of PacBio sequencing + barcoding in rationally arraying the
protein variants.

Can you do me a favour and test the code on influenza neuraminidase? Use
global H3N2 neuraminidase sequences from the IRD, starting from year 2000
to present. Under the assumption of only perfect matches, how many
originals out of the space of total possibles are there?


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/ericmjl/protein-systematic-characterization/pull/61#issuecomment-286457535>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ANrO2b95MzTEERSrquS5bdwuDEALU-Y3ks5rlrIFgaJpZM4LBsDb>.

ericmjl commented:
Great stuff! Now give it a shot on all sequences. If you want to offload the computation to the GPU workstation remotely, go for it!

Btw, if you're using a for-loop in your code, check out the Python package `tqdm`. It'll give you an estimate for the total loop time needed early on, which you can use to determine whether to offload or not.

Another btw: I think there's a matrix interpretation of path-finding between two nodes, even for directed acyclic graphs (which is what you're constructing). That might be really useful for speeding up computation. Check out this StackExchange question: http://math.stackexchange.com/questions/1890620/finding-path-lengths-by-the-power-of-adjacency-matrix-of-an-undirected-graph.

Starnite commented:
Thanks for the tips Eric! Is there a way to run the program in the GPU server without having to re-install all the packages?

ericmjl commented:
I think you can export your conda environment as a YAML file, and then use
the `conda create` command to re-create your compute environment on the GPU
workstation using the YAML file.

Check out the conda package manager docs:
https://conda.io/docs/using/envs.html

On Wed, Mar 15, 2017 at 11:52 AM, Starnite <notifications@github.com> wrote:

> Thanks for the tips Eric! Is there a way to run the program in the GPU
> server without having to re-install all the packages?
>
> From: Eric Ma <notifications@github.com>
> Reply-To: ericmjl/protein-systematic-characterization <
> reply@reply.github.com>
> Date: Wednesday, March 15, 2017 at 11:09 AM
> To: ericmjl/protein-systematic-characterization <protein-systematic-
> characterization@noreply.github.com>
> Cc: Vivian Zhong <vivzhong@mit.edu>, Mention <mention@noreply.github.com>
> Subject: Re: [ericmjl/protein-systematic-characterization] [WIP] tiled
> synthesis simulation (#61)
>
> Great stuff! Now give it a shot on all sequences. If you want to offload
> the computation to the GPU workstation remotely, go for it!Btw, if you're
> using a for-loop in your code, check out the Python package `tqdm`. It'll
> give you an estimate for the total loop time needed early on, which you can
> use to determine whether to offload or not.Another btw: I think there's a
> matrix interpretation of path-finding between two nodes, even for directed
> acyclic graphs (which is what you're constructing). That might be really
> useful for speeding up computation. Check out this StackExchange question:
> http://math.stackexchange.com/questions/1890620/finding-
> path-lengths-by-the-power-of-adjacency-matrix-of-an-undirected-graph.Sent
> from Nylas Mail, the best free email app for work
>
>
> On Mar 14 2017, at 10:42 pm, Starnite <notifications@github.com> wrote:
>
>
> Hi Eric,
>
>
>
> With tile lengths of 210 (30bp overlap on each end included), and testing
> with 10 sequences, I get 108 original out of 61666 total.
>
>
>
>
>
>
>
> From: Eric Ma <notifications@github.com>
>
> Reply-To: ericmjl/protein-systematic-characterization <
> reply@reply.github.com>
>
> Date: Tuesday, March 14, 2017 at 11:28 AM
>
> To: ericmjl/protein-systematic-characterization <protein-systematic-
> characterization@noreply.github.com>
>
> Cc: Vivian Zhong <vivzhong@mit.edu>, Mention <mention@noreply.github.com>
>
> Subject: Re: [ericmjl/protein-systematic-characterization] [WIP] tiled
> synthesis simulation (#61)
>
>
>
> Got it. This is interesting, it means that the pool of non-original
>
> sequences from the tiles is quite large, under the conditions where one
>
> mismatch per tile is allowed. This is good information; it'll elevate the
>
> importance of PacBio sequencing + barcoding in rationally arraying the
>
> protein variants.
>
>
>
> Can you do me a favour and test the code on influenza neuraminidase? Use
>
> global H3N2 neuraminidase sequences from the IRD, starting from year 2000
>
> to present. Under the assumption of only perfect matches, how many
>
> originals out of the space of total possibles are there?
>
>
>
>
>
> —
>
> You are receiving this because you were mentioned.
>
> Reply to this email directly, view it on GitHub<https://github.com/
> ericmjl/protein-systematic-characterization/pull/61#issuecomment-286457535>,
> or mute the thread<https://github.com/notifications/unsubscribe-auth/
> ANrO2b95MzTEERSrquS5bdwuDEALU-Y3ks5rlrIFgaJpZM4LBsDb>.
>
>
>
> —You are receiving this because you were mentioned.Reply to this email
> directly, view it on GitHub, or mute the thread.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub<https://github.com/
> ericmjl/protein-systematic-characterization/pull/61#issuecomment-286771926>,
> or mute the thread<https://github.com/notifications/unsubscribe-auth/
> ANrO2S6KVQ7BXUMOLfvnUVgzh_X0zklYks5rl_8ygaJpZM4LBsDb>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ericmjl/protein-systematic-characterization/pull/61#issuecomment-286786196>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACgnjrqUxuwBCepgD6vPuJDjXHW94OjNks5rmAkkgaJpZM4LBsDb>
> .
>

Starnite commented:
Hmm…I’m currently trying to use the ‘scp’ command to copy the yml file to the server, but I get “permission denied”.

ericmjl commented:
Hmm.... where are you copying to?

On Thu, Mar 16, 2017 at 10:33 PM, Starnite <notifications@github.com> wrote:

> Hmm…I’m currently trying to use the ‘scp’ command to copy the yml file to
> the server, but I get “permission denied”.
>
> From: Eric Ma <notifications@github.com>
> Reply-To: ericmjl/protein-systematic-characterization <
> reply@reply.github.com>
> Date: Wednesday, March 15, 2017 at 11:58 AM
> To: ericmjl/protein-systematic-characterization <protein-systematic-
> characterization@noreply.github.com>
> Cc: Vivian Zhong <vivzhong@mit.edu>, Mention <mention@noreply.github.com>
> Subject: Re: [ericmjl/protein-systematic-characterization] [WIP] tiled
> synthesis simulation (#61)
>
> I think you can export your conda environment as a YAML file, and then use
> the `conda create` command to re-create your compute environment on the
> GPU
> workstation using the YAML file.
>
> Check out the conda package manager docs:
> https://conda.io/docs/using/envs.html
>
> On Wed, Mar 15, 2017 at 11:52 AM, Starnite <notifications@github.com>
> wrote:
>
> > Thanks for the tips Eric! Is there a way to run the program in the GPU
> > server without having to re-install all the packages?
> >
> > From: Eric Ma <notifications@github.com>
> > Reply-To: ericmjl/protein-systematic-characterization <
> > reply@reply.github.com>
> > Date: Wednesday, March 15, 2017 at 11:09 AM
> > To: ericmjl/protein-systematic-characterization <protein-systematic-
> > characterization@noreply.github.com>
> > Cc: Vivian Zhong <vivzhong@mit.edu>, Mention <mention@noreply.github.com>
>
> > Subject: Re: [ericmjl/protein-systematic-characterization] [WIP] tiled
> > synthesis simulation (#61)
> >
> > Great stuff! Now give it a shot on all sequences. If you want to offload
> > the computation to the GPU workstation remotely, go for it!Btw, if
> you're
> > using a for-loop in your code, check out the Python package `tqdm`.
> It'll
> > give you an estimate for the total loop time needed early on, which you
> can
> > use to determine whether to offload or not.Another btw: I think there's
> a
> > matrix interpretation of path-finding between two nodes, even for
> directed
> > acyclic graphs (which is what you're constructing). That might be really
> > useful for speeding up computation. Check out this StackExchange
> question:
> > http://math.stackexchange.com/questions/1890620/finding-
> > path-lengths-by-the-power-of-adjacency-matrix-of-an-undirected-graph.Sent
>
> > from Nylas Mail, the best free email app for work
> >
> >
> > On Mar 14 2017, at 10:42 pm, Starnite <notifications@github.com> wrote:
> >
> >
> > Hi Eric,
> >
> >
> >
> > With tile lengths of 210 (30bp overlap on each end included), and
> testing
> > with 10 sequences, I get 108 original out of 61666 total.
> >
> >
> >
> >
> >
> >
> >
> > From: Eric Ma <notifications@github.com>
> >
> > Reply-To: ericmjl/protein-systematic-characterization <
> > reply@reply.github.com>
> >
> > Date: Tuesday, March 14, 2017 at 11:28 AM
> >
> > To: ericmjl/protein-systematic-characterization <protein-systematic-
> > characterization@noreply.github.com>
> >
> > Cc: Vivian Zhong <vivzhong@mit.edu>, Mention <mention@noreply.github.com>
>
> >
> > Subject: Re: [ericmjl/protein-systematic-characterization] [WIP] tiled
> > synthesis simulation (#61)
> >
> >
> >
> > Got it. This is interesting, it means that the pool of non-original
> >
> > sequences from the tiles is quite large, under the conditions where one
> >
> > mismatch per tile is allowed. This is good information; it'll elevate
> the
> >
> > importance of PacBio sequencing + barcoding in rationally arraying the
> >
> > protein variants.
> >
> >
> >
> > Can you do me a favour and test the code on influenza neuraminidase? Use
> >
> > global H3N2 neuraminidase sequences from the IRD, starting from year
> 2000
> >
> > to present. Under the assumption of only perfect matches, how many
> >
> > originals out of the space of total possibles are there?
> >
> >
> >
> >
> >
> > —
> >
> > You are receiving this because you were mentioned.
> >
> > Reply to this email directly, view it on GitHub<https://github.com/
> > ericmjl/protein-systematic-characterization/pull/61#issuecomment-286457535>,
>
> > or mute the thread<https://github.com/notifications/unsubscribe-auth/
> > ANrO2b95MzTEERSrquS5bdwuDEALU-Y3ks5rlrIFgaJpZM4LBsDb>.
> >
> >
> >
> > —You are receiving this because you were mentioned.Reply to this email
> > directly, view it on GitHub, or mute the thread.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub<https://github.com/
> > ericmjl/protein-systematic-characterization/pull/61#issuecomment-286771926>,
>
> > or mute the thread<https://github.com/notifications/unsubscribe-auth/
> > ANrO2S6KVQ7BXUMOLfvnUVgzh_X0zklYks5rl_8ygaJpZM4LBsDb>.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/ericmjl/protein-systematic-characterization/pull/61#
> issuecomment-286786196>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/
> ACgnjrqUxuwBCepgD6vPuJDjXHW94OjNks5rmAkkgaJpZM4LBsDb>
> > .
> >
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub<https://github.com/
> ericmjl/protein-systematic-characterization/pull/61#issuecomment-286788517>,
> or mute the thread<https://github.com/notifications/unsubscribe-auth/
> ANrO2Wzr8ygw1yKhZFljpyo63g27rccbks5rmAq1gaJpZM4LBsDb>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ericmjl/protein-systematic-characterization/pull/61#issuecomment-287250398>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACgnjiI63SQWF8qSb3-FSktBZ2I9Q6wCks5rmfD5gaJpZM4LBsDb>
> .
>

Starnite commented:
vivzhong@ubuntu-gpu.myddns.me

ericmjl commented:
Let me try resetting your pw.

ericmjl commented:
Try again. Your pw is vivianzhong@mit.edu.

To change your pw, after you login, use "passwd vivzhong", and quickly
change it.

Let's see if that fixes the permissions error.

Finally, just checking, you aren't scp-ing to the root directory, but your
home directory right? The destination should start with "~/".

On Thu, Mar 16, 2017 at 11:00 PM, Eric Ma <ericmajinglong@gmail.com> wrote:

> Let me try resetting your pw.
>

Starnite commented:
Thanks Eric! So I managed to scp the environment.yml file into the home directory, but it seems like I’d still need to do some installing with sudo<http://datascienceguide.github.io/how-to-install-the-python-data-science-stack-on-a-remote-server> and pip in order to use the conda command, and I don’t have sudo privileges.

ericmjl commented:
Great progress! I think you can use the Anaconda distribution of Python to
get around needing to use sudo privileges.
On Fri, Mar 17, 2017 at 5:42 PM Starnite <notifications@github.com> wrote:

> Thanks Eric! So I managed to scp the environment.yml file into the home
> directory, but it seems like I’d still need to do some installing with sudo<
> http://datascienceguide.github.io/how-to-install-the-python-data-science-stack-on-a-remote-server>
> and pip in order to use the conda command, and I don’t have sudo
> privileges.
>
>
> From: Eric Ma <notifications@github.com>
> Reply-To: ericmjl/protein-systematic-characterization <
> reply@reply.github.com>
> Date: Thursday, March 16, 2017 at 11:03 PM
> To: ericmjl/protein-systematic-characterization <
> protein-systematic-characterization@noreply.github.com>
> Cc: Vivian Zhong <vivzhong@mit.edu>, Mention <mention@noreply.github.com>
> Subject: Re: [ericmjl/protein-systematic-characterization] [WIP] tiled
> synthesis simulation (#61)
>
> Try again. Your pw is vivianzhong@mit.edu.
>
> To change your pw, after you login, use "passwd vivzhong", and quickly
> change it.
>
> Let's see if that fixes the permissions error.
>
> Finally, just checking, you aren't scp-ing to the root directory, but your
> home directory right? The destination should start with "~/".
>
> On Thu, Mar 16, 2017 at 11:00 PM, Eric Ma <ericmajinglong@gmail.com>
> wrote:
>
> > Let me try resetting your pw.
> >
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub<
> https://github.com/ericmjl/protein-systematic-characterization/pull/61#issuecomment-287254394>,
> or mute the thread<
> https://github.com/notifications/unsubscribe-auth/ANrO2QvVVifemJQH8QczBHZ3bWB0HuJNks5rmff9gaJpZM4LBsDb>.
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ericmjl/protein-systematic-characterization/pull/61#issuecomment-287478749>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACgnjlIsH3t59lnF5ycI5v7q2G-RBC6iks5rmv5hgaJpZM4LBsDb>
> .
>

Starnite commented:
I have everything working! But the code as is definitely takes far too long to run, so I’ll work on optimizing it. With 50 starting sequences, we get 4337 original and 290651837 new sequences.

ericmjl commented:
Good stuff! I notice that your code is particularly useful for estimating the proportion of "desired" sequences that can show up. You might want to do a Monte Carlo walk on the graph, as an alternative in case optimization takes a long time.

Starnite commented:
Hey Eric,

Sorry for falling off the grid! I tried to do what I tried to do what I thought would be a simple simplification, but it ended up making the runtime even longer.

My initial method of adding edges was to simply evaluate every pairing of tiles in the digraph as determined by itertools.product. I figured that this was pretty unnecessary, and changed the code so that it only compared two tiles if the second came right after the first in the DNA sequence. This actually resulted in more edges being added, and thus exponentially more paths to search and exponentially longer runtime.

For example, with 2 sequences, 8 tiles each (for the neuraminidase):

First method:
num nodes: 16
num edges: 19
original: 2 new: 5
Second method:
num nodes: 16
num edges: 24
original: 2 new: 62

The case where every overlap is perfect should result in 28 edges (2 edges per node in a sequence * 7 nodes in a sequence (excluding the last one) * 2 sequences), so I’m inclined to think that the second method is correct. I’m not sure why the first method lost a few edges, though – perhaps itertools.product isn’t getting everything?

Starnite commented:
Addendum: I checked the runtime using the second method for a few different numbers of sequences, and extrapolating from a fitted exponential curve, the code as is would take 68 days to sort through ~2000 sequences. So I’ll definitely look into random walk MC.

ericmjl commented:
Hmm... It looks like it'll be worth us meeting up to look through some code. I have a suspicion that you'll need to incorporate more test cases in test suite, but let's chat in person. Got a time that works next week?

Starnite commented:
Hi Eric,

Would Thursday 2:30 work for you? I unfortunately have an exam on Wednesday.

I know you mentioned earlier this semester that you wanted to record me carrying out the 96-well polymerase assay; do you have an approximate date in mind for that?

ericmjl commented:
@Starnite Thursday's good; I'll be at the Sloan School of Mgmt all day though (there's a 1-day conference going on there), will it be okay with you to walk over there? If not, we can always do a Hangout call after hours too.

ericmjl commented:
As for videographing the 96-well plate assay, let me work out some other experimental details; there's still one transfection experiment I'll need your help with as well, but I need to make sure the protocol we have is good to go.

Starnite commented:
Hi Eric,

My 6.036 project deadline just got pushed to Monday, so I’d be happy to make the trip. :) What time works for you?

ericmjl commented:
@Starnite I took a look at the agenda for the day, it looks like everything ends at 4 pm. Do you think that's a good time for you? I originally thought 2:30 pm, but as it turns out there's only a 15 min. break there -_-", but I think your stuff is worth spending a good hour (or more) on.

Starnite commented:
Yup, 4 is good!





On Wed, Apr 5, 2017 at 10:49 PM -0400, "Eric Ma" <notifications@github.com<mailto:notifications@github.com>> wrote:


@Starnite<https://github.com/Starnite> I took a look at the agenda for the day, it looks like everything ends at 4 pm. Do you think that's a good time for you? I originally thought 2:30 pm, but as it turns out there's only a 15 min. break there -_-", but I think your stuff is worth spending a good hour (or more) on.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/ericmjl/protein-systematic-characterization/pull/61#issuecomment-292054374>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ANrO2afNmi9DabemTjSlTTF3cezEfxi3ks5rtFLBgaJpZM4LBsDb>.

ericmjl commented:
ping @Starnite: as promised yesterday, let's merge this and start a new PR to continue the conversation!
